src/finalfusion/io.py

```diff
@@ -0,0 +1,280 @@
+        bytes are returned.
+
+        Parameters
+        ----------
+        file : IO[bytes]
+            a finalfusion file positioned before a chunk header.
+
+        Returns
+        -------
+        chunk_header : Optional[(ChunkIdentifier, int)]
+            None is returned iff the reader is at EOF.
+
+        Raises
+        ------
+        FinalfusionFormatError
+            If only part of the header could be read.
+        """
+        val = Chunk._read_binary(file, "<IQ")
+        if val is None:
+            return None
+        return ChunkIdentifier(val[0]), val[1]
+
+    @staticmethod
+    def _write_binary(file: IO[bytes], struct_fmt: str, *args):
+        """
+        Helper method to write binary data according to the format string.
+        :param file:
+        :param struct_fmt:
+        :param args:
+        :return:
+        """
+        data = struct.pack(struct_fmt, *args)
+        file.write(data)
+
+    @staticmethod
+    def _read_binary(file: IO[bytes], struct_fmt: str) -> Optional[Tuple[int]]:
+        """
+        Helper method to read binary data from a file according to the format
+        string.
+
+        Parameters
+        ----------
+        file : IO[bytes]
+            Output file
+        struct_fmt : str
+            struct format string
+
+        Returns
+        -------
+        data : Optional[tuple]
+            Returns the unpacked data as a tuple. If **no** data could be read,
+            None is returned
+
+        Raises
+        ------
+        FinalfusionFormatError
+            If data could only be read partially.
+        """
+        size = struct.calcsize(struct_fmt)
+        buf = file.read(size)
+        if len(buf) == 0:
+            return None
+        if len(buf) != size:
+            raise FinalfusionFormatError(
+                "Could not read {} bytes from file".format(size))
+        return struct.unpack(struct_fmt, buf)
+
+
+class Header(Chunk):
+    """
+    Header Chunk
+
+    The header chunk handles the preamble.
+    """
+    def __init__(self, chunk_ids):
+        self.chunk_ids_ = chunk_ids
+
+    @property
+    def chunk_ids(self) -> List['ChunkIdentifier']:
+        """
+        Get the chunk IDs from the header
+
+        Returns
+        -------
+        chunk_ids : List[ChunkIdentifier]
+            List of ChunkIdentifiers in the Header.
+        """
+        return self.chunk_ids_
+
+    @staticmethod
+    def chunk_identifier() -> 'ChunkIdentifier':
+        return ChunkIdentifier.Header
+
+    @staticmethod
+    def read_chunk(file) -> 'Header':
+        magic = file.read(4)
+        if magic != _MAGIC:
+            raise FinalfusionFormatError(
+                "Magic should be b'FiFu', not: " +
+                magic.decode('utf-8', errors='ignore'))
+        version = Header._read_binary(file, "<I")[0]
+        if version != VERSION:
+            raise FinalfusionFormatError("Unknown model version: " +
```

Comment:
f'Unknown model version: {str(version)}'

I guess this applies for the other string concatenations as well. If it becomes a bit unwieldy, I think

"foo {}".format(bar)

is still nicer than concatenation.